packages/block-library/src/social-links/edit.native.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import { InnerBlocks } from '@wordpress/block-editor';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { useRef, useEffect, useState } from '@wordpress/element';
+import { compose, usePreferredColorSchemeStyle } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import variations from '../social-link/variations';
+
+const ALLOWED_BLOCKS = variations.map(
+	( v ) => `core/social-link-${ v.name }`
+);
+
+// Template contains the links that show when start.
+const TEMPLATE = [
+	[
+		'core/social-link-wordpress',
+		{ service: 'wordpress', url: 'https://wordpress.org' },
+	],
+	[ 'core/social-link-facebook', { service: 'facebook' } ],
+	[ 'core/social-link-twitter', { service: 'twitter' } ],
+	[ 'core/social-link-instagram', { service: 'instagram' } ],
+];
+
+function SocialLinksEdit( {
+	shouldDelete,
+	onDelete,
+	isSelected,
+	isInnerIconSelected,
+	innerBlocks,
+	attributes,
+	activeInnerBlocks,
+	getBlock,
+} ) {
+	const [ initialCreation, setInitialCreation ] = useState( true );
+	const shouldRenderFooterAppender = isSelected || isInnerIconSelected;
+	const { align } = attributes;
+	const { marginLeft: spacing } = styles.spacing;
+
+	useEffect( () => {
+		if ( ! shouldRenderFooterAppender ) {
+			setInitialCreation( false );
+		}
+	}, [ shouldRenderFooterAppender ] );
+
+	const renderFooterAppender = useRef( () => (
+		<View>
+			<InnerBlocks.ButtonBlockAppender isFloating={ true } />
+		</View>
+	) );
+
+	const placeholderStyle = usePreferredColorSchemeStyle(
+		styles.placeholder,
+		styles.placeholderDark
+	);
+
+	function renderPlaceholder() {
+		return [ ...new Array( innerBlocks.length || 1 ) ].map( ( index ) => (
```

Comment:
We shouldn't use the index as a key but I imagine that in some cases this is impossible to avoid, however, I believe it should look like this :)
```js
.map( ( _, index ) => (
```

Reply:
I can agree that name `index` is misleading and maybe I should call it just `e` or `item` since it's only an array of numbers.